src/main/java/br/com/caelum/vraptor/jpa/EntityManagerFactoryCreator.java

```diff
@@ -33,15 +37,52 @@
  * @author Otávio Garcia
  */
 public class EntityManagerFactoryCreator {
-	
-	@Inject 
+	@Inject
 	private Environment environment;
+	@Inject
+	private Properties propertiesOfJPAConnection;
 
+	
+	/**
+	 * Produces EntityManagerFactory.  For default  is created an connect 
+	 * based on configurations of the persistence.xml. 
+	 * However, if the database properties are unset in persistence.xml,  
+	 * the connection will created programmatically by propertiesOfJPAConnection,
+	 * overriding the default settings.
+	 * 
+	 * @return the EntityManagerFactory that will create all Entity managers
+	 * @throws IOException 
+	 * @throws FileNotFoundException 
+	 */
 	@ApplicationScoped
 	@Produces
-	public EntityManagerFactory getEntityManagerFactory() {
+	public EntityManagerFactory getEntityManagerFactory(){
+		EntityManagerFactory factory = null;
 		String persistenceUnit = environment.get("br.com.caelum.vraptor.jpa.persistenceunit", "default");
-		return Persistence.createEntityManagerFactory(persistenceUnit);
+		
+		if(!propertiesOfJPAConnection.isEmpty() && !hasXMLConfiguration(persistenceUnit)) {
+			factory = Persistence.createEntityManagerFactory(persistenceUnit, propertiesOfJPAConnection);
+		} else {
+			factory = Persistence.createEntityManagerFactory(persistenceUnit);
+		}
+		return factory;			
+	}
+	
+	/**
+	 * Checks if exist properties of persistence unit in persistence.xml
+	 * 
+	 * @param persistenceUnit to validate
+	 * @return true if settings exist, this not checks if are invalids
+	 */
+	private boolean hasXMLConfiguration(String persistenceUnit){
+		final String url = "hibernate.connection.url";
```

Comment:
This don't works, because:
1) Only works for Hibernate, and as a JPA bridge we need to cover all JPA implementations like KODO, OpenJPA, Toplink and EclipseLink. 
2) There is several ways to configure Hibernate. If you are using datasource, this property is not valid.